src/debug-generator/debug-generator.c

```diff
@@ -142,6 +159,32 @@ static int generate_wants_symlinks(void) {
         return r;
 }
 
+static int install_debug_shell_dropin(const char *tty) {
+        const char *dropin = "/run/systemd/system/debug-shell.service.d/50-tty.conf";
+        const char *dir = "/run/systemd/system/debug-shell.service.d/";
+        _cleanup_free_ char *text = NULL;
+        int r;
+
+        r = mkdir_p(dir, 0755);
+        if (r < 0 && r != -EEXIST)
+                return log_debug_errno(r, "Failed to create directory '%s': %m", dir);
+
+        r = asprintf(&text, "[Unit]\n"
+                            "Description=Early root shell on /dev/%s FOR DEBUGGING ONLY\n"
+                            "ConditionPathExists=/dev/%s\n"
```

Comment:
i would probably drop this condition check here, since it probably is better to fail if we are explicitly told to use some tty than to silenelty eat it up.

moreover, conditions are maintained in lists, you hence should reset the list first, i.e. assign ConditionPathExists= the empty string first.

Reply:
The ConditionPathExists= check is currently in units/debug-shell.service.in. Should I remove it from there as well?

Reply:
Yeah. `debug-shell.service` is not enabled by default, so it should fail if enabled and unable to do its job.